src/custom/components/SearchModal/CurrencyList/CurrencyListMod.tsx

```diff
@@ -273,14 +274,19 @@ export default function CurrencyList({
   BalanceComponent?: (params: { balance: CurrencyAmount<Currency> }) => JSX.Element // gp-swap added
   TokenTagsComponent?: (params: { currency: Currency; isUnsupported: boolean }) => JSX.Element // gp-swap added
 }) {
+  const allTokens = useAllTokens()
   const isUnsupportedToken = useIsUnsupportedTokenGp()
 
   const itemData: (Currency | BreakLine)[] = useMemo(() => {
     if (otherListTokens && otherListTokens?.length > 0) {
-      return [...currencies, BREAK_LINE, ...otherListTokens]
+      const filteredOtherListTokens = otherListTokens.filter((token) =>
+        token.isToken ? !allTokens[token.address.toLowerCase()] : true
```

Comment:
nit: maybe add a tiny comment?

Reply:
Thanks!
Added a comment and added TODO to refactor the component